src/cloudevent/datamodel/ucloudeventattributes.rs

```diff
@@ -69,7 +39,7 @@ impl UCloudEventAttributes {
         &self.hash
     }
 
-    pub fn priority(&self) -> &Option<Priority> {
+    pub fn priority(&self) -> &Option<UPriority> {
```

Comment:
In general it's better to return `Option<&T>` instead of `&Option<T>`.
If the function returns `Option<&T>`, then `T` has the freedom to live either on the heap or on the stack (like in this case). I.e., `T` or `Box<T>` can be both referenced with `&T`.
Instead, if the function returns `&Option<T>`, then `T` needs to live on the stack. If you want to move it to the heap (e.g. box-it), you will need to change the function signature to `&Option<Box<T>>`.

Reply:
Done